cli/maintainer/src/maintenance.rs

```diff
@@ -754,33 +768,108 @@ impl SolidoState {
             .exchange_sol(Lamports(anker_state.b_sol_total_supply_amount.0))
             .expect("It will not overflow because we always have less than the total amount of minted Sol.");
 
-        let rewards = (reserve_st_sol - st_sol_amount).ok()?;
+        (reserve_st_sol - st_sol_amount).ok()
+    }
 
+    /// Try to update the historical pool price exchange rate to protect us
+    /// against sandwiching attacks.
+    /// We try to update the historical price when we're near the epoch
+    /// boundaries.
+    pub fn try_fetch_pool_price(&self) -> Option<MaintenanceInstruction> {
```

Comment:
I think we can simplify this and keep only the `try_sell_anker_rewards` function. Or at least, keep only that function in the `.or_else` chain at line 1688. `try_sell_anker_rewards` would then do something like this (pseudocode):

```rust
if rewards == 0 {
    return None;
}

// Confirm that the price samples are up to date before we sell.
if oldest_price_sample.age() > MAX_AGE {
    if youngest_price_sample.age() < MIN_DISTANCE {
        // Too soon, can’t update or sell yet.
        // Possibly we would need something to not make it wait
        // too long before the next iteration.
        return None; 
    }
    
    return call_fetch_pool_price();
}

// If we get here, the price samples are up to date and we can sell,
// but it might still be too soon.
if youngest_price_sample.age() < MIN_DISTANCE {
    return None;
}

return call_sell_rewards();
```

It does create some delay versus fetching the price at the end of the previous epoch already, but I don’t think that’s a problem, the exact time we sell the rewards doesn’t matter as long as we can do it once per epoch. Also, if somebody else starts running a cranker bot and they do update the price at the end of the previous epoch ... good for them, then we can immediately sell the rewards.